### PR TITLE
Removed same text on comment reply form

### LINKED ISSUE #9500: When replying to a newly posted comment, the last comment text appears in the new "reply" comment form
This is on the non-React version so it may simply be solved by adopting the React version, but we'd have to test that it doesn't occur there. 

See how i just posted this comment and when i try to reply to the new comment, i see this in the textarea:

image

### PR DESCRIPTION
**Pull request by waridrox**
Fixes #9500 and #9508 
Removed same text on first comment reply

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by RuthNjeri** [comment]
Thanks, @waridrox, is it possible to add a test for this?

**Comment by waridrox** [comment]
Sure, @RuthNjeri I'll add one 😄 but first I wanted to confirm if [this](https://github.com/publiclab/plots2/issues/9500#issuecomment-819879573-permalink:~:text=Can%20someone%20pls%20confirm%20if%20the,it%20just%20specific%20to%20local%20env...thanks.) problem persists on production env, which may bring some modifications to the above changes. Thanks :)

**Comment by RuthNjeri** [comment]
Alright, thanks @waridrox. I think you can test it on the stable environment first...there was a way to push directly to stable, @jywarren, is this something @waridrox can do?

**Comment by jywarren** [comment]
Hi, actually you need push rights to publiclab/plots2 `unstable` branch, and then it publishes at http://unstable.publiclab.org/. @RuthNjeri i believe you have these privileges but if you don't get to it by tomorrow i can do it! However, we're pretty satisfied with either a system test or a manual test in GitPod, if that's easier?

Thanks!

**Comment by RuthNjeri** [comment]
Thanks @jywarren, I pushed your changes to the [unstable branch](https://github.com/publiclab/plots2/compare/unstable) @waridrox, you can test your changes in the link Jeff shared for the stable branch

**Comment by waridrox** [comment]
Thanks a lot @RuthNjeri for the help, but I already tested this issue on the official website [here](https://github.com/publiclab/plots2/issues/9500#issuecomment-827477504) and it is persistent there, should I also try on the unstable branch then...?

**Comment by RuthNjeri** [comment]
@waridrox, the unstable site has the changes for this PR, to understand you better, are you saying that the current changes do not fix the initial problem that was also happening on the main website?

**Comment by waridrox** [comment]
> @waridrox, the unstable site has the changes for this PR, to understand you better, are you saying that the current changes do not fix the initial problem that was also happening on the main website?

I'm sorry if I puzzled you 😅, but in an attempt to solve the main issue, i.e.
1 - removing reply text on the first reply to a comment
there was another issue that was observed after solving the above, i.e.
2 - After replying once on a comment, if we reply again, the contents of the previous comment reply form are copied to a new comment reply form even if reply to a comment button is not clicked.

File: app/views/comments/_form.html.erb
Comment (jywarren) [inline app/views/comments/_form.html.erb:95]: Hmm, i guess we need a different solution. What was the original intent of this phrase?

```ruby
location == :edit ? comment.comment : params[:body]
```

Is it to see if the location is "edit" and if it is, display the existing comment text, but if not, to display the value of the `params[:body]` variable? 

I wonder if it makes the assumption that the `params[:body]` variable is corresponding to... the submitted content of a comment, and if we should be checking to ensure it's the correct value in some other way? Remember that this may be used both on initial pageload and also in an AJAX response, which complicates this. 

It could be useful to monitor the network pane of the Dev Tools to see if the partial template being sent from the server to generate this textarea is bearing the correct content?

**Comment by waridrox** [comment]
@jywarren @RuthNjeri, so I tried to investigate the problem further, figuring out where the data gets passed on after making a comment, and I realised that the text input field’s value was not cleared after making a POST AJAX request along with a NotyNotification! 😅. That's why the same text was repeating the comment reply form...

So, resetting the text input field value resolved the issue: 


https://user-images.githubusercontent.com/58583793/117176100-23897f00-aded-11eb-8ec9-e2c52cf94c93.mp4

**Comment by jywarren** [comment]
Oh wow awesome!

Hmm, is the system test failure unrelated? Shall we restart it?

**Comment by waridrox** [comment]
> Oh wow awesome!
> 
> Hmm, is the system test failure unrelated? Shall we restart it?

Screenshot 2021-05-06 at 2 56 10 AM

@jywarren, I suppose they need to be updated to take into account the new change because they were written when the contents of the reply were repeated 😅. Also, I think these tests will have some effect if #9384 got merged...

**Comment by waridrox** [comment]
So an edge case which I forgot to thought of 😅. Clearing the contents of all text input fields meant that editing the comment was not possible, but with the new changes it does. I'm not able to figure out though why this test is still failing 😕. @jywarren @cesswairimu @RuthNjeri, need some help...- 
Screenshot 2021-05-12 at 1 18 49 AM

failures_test_note:_multiple_comment_boxes,_post_comments

Once this is fixed, all tests will pass :)

**Comment by waridrox** [comment]
Ok so finally figured out why the tests were failing, the `textarea` content value was getting cleared when the reply to individual comments were published. Fixed that and the tests pass now :) @jywarren pls have a look!

https://user-images.githubusercontent.com/58583793/118195338-a901d480-b468-11eb-863f-63de4592a2e9.mp4

**Review by pydevsg — APPROVED** [review]
Good work @waridrox 🎉

**Comment by RuthNjeri** [comment]
Thanks for working on this @waridrox and great job investigating the edge cases. Would you like to write a test for this? I think it will be useful for anyone to understand what is expected of the code written. You can also set it up as another issue so that we merge this one first if you would like.

**Review by RuthNjeri — APPROVED** [review]
LGTM: just a pending test.

**Comment by waridrox** [comment]
> LGTM: just a pending test.

Sure thing! 🚀

**Comment by jywarren** [comment]
Awesome work. Thank you so much!!!!!!

**Pull request merged**
